firebolt_provider/hooks/firebolt.py

```diff
@@ -86,36 +107,96 @@ def _get_conn_params(self) -> Dict[str, Optional[str]]:
             "password": conn.password or "",
-            "api_endpoint": conn.host or DEFAULT_API_URL,
+            "api_endpoint": DEFAULT_API_URL,
             "database": self.database or database,
-            "engine_name": self.engine_name or engine_name,
+            "engine_name": engine_name,
+            "engine_url": engine_url,
             "account_name": account_name,
         }
         return conn_config
 
     def get_conn(self) -> Connection:
         """Return Firebolt connection object"""
         conn_config = self._get_conn_params()
-        conn = connect(**conn_config)
+        username, password = conn_config["username"], conn_config["password"]
+        if not (username and password):
+            raise FireboltError("Either username or password is missing")
+
+        conn = connect(
+            auth=UsernamePassword(username=username, password=password),
+            api_endpoint=conn_config["api_endpoint"],
+            database=conn_config["database"],
+            engine_name=conn_config["engine_name"],
+            engine_url=conn_config["engine_url"],
+            account_name=conn_config["account_name"],
+        )
+
         return conn
 
     def get_resource_manager(self) -> ResourceManager:
         """Return Resource Manager"""
         conn_config = self._get_conn_params()
-        return ResourceManager(
+        username, password = conn_config["username"], conn_config["password"]
+        if not (username and password):
+            raise FireboltError("Either username or password is missing")
+
+        manager = ResourceManager(
             Settings(
-                user=conn_config["username"],
-                password=conn_config["password"],
+                auth=UsernamePassword(username=username, password=password),
                 server=conn_config["api_endpoint"],
                 default_region="us-east-1",
             )
         )
 
+        return manager
+
+    def engine_action(self, engine_name: Optional[str], action: str) -> None:
+        """
+        Performs start or stop of the engine
+
+        Args:
+            engine_name: name of the engine, if None, the engine from
+             the connection will be used
+            action: either stop or start
+
+        """
+        if engine_name is None:
+            self.log.info(
+                "engine_name is not set, getting engine_name from connection config"
+            )
+            conn_config = self._get_conn_params()
+            database_name = conn_config.get("database")
+
+            engine_name = conn_config.get("engine_name", None)
+            engine_url = conn_config.get("engine_url", None)
+            if engine_url:
+                raise FireboltError("Cannot start/stop engine using its URL")
+
+        rm = self.get_resource_manager()
+        if engine_name is None:
+            if database_name is None:
+                raise FireboltError("Database cannot be not set")
```

Comment:
nit: Double negative, I'd suggest "Database must be set".

Reply:
Done